src/core/components/states/stateSkeletonBar/stateSkeletonBar.tsx

```diff
@@ -0,0 +1,49 @@
+import classNames from 'classnames';
+import { forwardRef, type ComponentPropsWithoutRef } from 'react';
+import { type ResponsiveAttribute, type ResponsiveAttributeClassMap } from '../../../types';
+import { responsiveUtils } from '../../../utils';
+
+export type SkeletonBarSize = 'sm' | 'md' | 'lg' | 'xl' | '2xl';
+
+export interface IStateSkeletonBarProps extends ComponentPropsWithoutRef<'span'> {
+    /**
+     *  Responsive size attribute for the skeleton.
+     */
+    responsiveSize?: ResponsiveAttribute<SkeletonBarSize>;
+    /**
+     * The size of the skeleton.
+     * @default 'md'
+     */
+    size?: SkeletonBarSize;
+}
+
+const responsiveSizeClasses: ResponsiveAttributeClassMap<SkeletonBarSize> = {
+    sm: { sm: 'h-3', md: 'md:h-3', lg: 'lg:h-3', xl: 'xl:h-3', '2xl': '2xl:h-3' },
+    md: { sm: 'h-4', md: 'md:h-4', lg: 'lg:h-4', xl: 'xl:h-4', '2xl': '2xl:h-4' },
+    lg: { sm: 'h-5', md: 'md:h-5', lg: 'lg:h-5', xl: 'xl:h-5', '2xl': '2xl:h-5' },
+    xl: { sm: 'h-6', md: 'md:h-6', lg: 'lg:h-6', xl: 'xl:h-6', '2xl': '2xl:h-6' },
+    '2xl': { sm: 'h-8', md: 'md:h-8', lg: 'lg:h-8', xl: 'xl:h-8', '2xl': '2xl:h-8' },
+};
+
+export const StateSkeletonBar = forwardRef<HTMLDivElement, IStateSkeletonBarProps>((props, ref) => {
+    const { className, responsiveSize = {}, size = 'md', ...otherProps } = props;
+
+    const classes = classNames(
+        'w-40 animate-pulse rounded-full bg-neutral-50',
```

Comment:
What's the plan for customising the width of the component?

Reply:
This is just a default width so the component does appear by default. When it's being used, we then override using classnames.

Reply:
Mm.. Have you tried overriding it on a Storybook story? I'm asking because we might need to always use the important modifier if we want to change the bar width which is not optimal. An alternative would be using a `width` attribute which set the width as inline style?

Reply:
Ah, right because no real idea in what order tailwind actually defines the actual CSS classes. `width` is a better approach since it doesn't make sense to have the component have zero width by default; not a huge fan because we essentially have 3 different ways to mess with the sizing of the component.